src/library_webgl.js

```diff
@@ -3794,19 +3794,37 @@ for (/**@suppress{duplicate}*/var i = 0; i <= {{{ GL_POOL_TEMP_BUFFERS_SIZE }}};
   glDrawElements: (mode, count, type, indices) => {
 #if FULL_ES2
     var buf;
+    var vertexes = 0;
     if (!GLctx.currentElementArrayBufferBinding) {
       var size = GL.calcBufLength(1, type, 0, count);
       buf = GL.getTempIndexBuffer(size);
       GLctx.bindBuffer(0x8893 /*GL_ELEMENT_ARRAY_BUFFER*/, buf);
       GLctx.bufferSubData(0x8893 /*GL_ELEMENT_ARRAY_BUFFER*/,
                           0,
                           HEAPU8.subarray(indices, indices + size));
+      
+      // Calculating vertex count if shader's attribute data is on client side
+      if (count > 0) {
+        for (var i = 0; i < GL.currentContext.maxVertexAttribs; ++i) {
+          var cb = GL.currentContext.clientBuffers[i];
+          if (cb.clientside && cb.enabled) {
+            var array_classes = {
+                5121 /* GL_UNSIGNED_BYTE */: Uint8Array,
+                5123 /* GL_UNSIGNED_SHORT */: Uint16Array,
+                5125 /* GL_UNSIGNED_INT */: Uint32Array};
+            if (array_classes[type])
```

Comment:
What happens if the type is not one of the known classes? Should we not error?

Reply:
Per GL spec it would be correct for a GL_INVALID_ENUM error to be raised, so that execution can continue. Although given that this is on a rare emulation path, I think it would be ok to let this throw an exception on the next line.

Reply:
Handling for unknown types has been added.